#187419000 Configure ESLint and Git Pre commit hook  And Hound CI

JacquelineTuyisenge opened the pull request:
#### Configure ESLint and Git Pre commit hook to run tests and ESlint format before commits And Hound CI

#### How should this be manually tested?

 Clone the repository branch in the local Machine
 perform npm install to have all packages
 Change something in a  files and commit
  See how the configured tools works, like not using imported files and declaring variables and not use them

#### What are the relevant pivotal tracker/Trello stories?

#187419000

JacquelineTuyisenge committed:
eslint

JacquelineTuyisenge committed:
eslint

JacquelineTuyisenge committed:
eslint

JacquelineTuyisenge committed:
chore(setup Eslint, Pre-commit and Hound ci): setting up the eslint, precommit and hound ci

JacquelineTuyisenge committed:
chore(Eslint): Eslint, git preccommit and hound

leandreAlly commented on package.json:
```diff
@@ -25,9 +32,15 @@
+    "eslint-config-prettier": "9.1.0",
+    "eslint-plugin-react": "^7.34.2",
+    "eslint-plugin-react-hooks": "^4.6.2",
+    "eslint-plugin-react-refresh": "^0.4.7",
```
install all this package start with eslint ad dev dependency.

Hakizimana-Clement committed:
test(testing folder): Change folder structure to be in the root folder

- Moved the test folder from src to the root folder for better organization.
- Added setupTest.ts in the test folder.
- Updated vite.config.ts to include the test configuration.

JacquelineTuyisenge committed:
my commit

JacquelineTuyisenge committed:
my commit
